crates/sui-json-rpc/src/read_api.rs

```diff
@@ -638,6 +638,27 @@ impl ReadApiServer for ReadApi {
         })
     }
 
+    #[instrument(skip(self))]
+    async fn try_get_object_before_version(
+        &self,
+        object_id: ObjectID,
+        version: SequenceNumber,
+    ) -> RpcResult<SuiPastObjectResponse> {
+        let version = self
+            .state
+            .find_object_lt_or_eq_version(&object_id, &version)
+            .await
+            .map_err(Error::from)?
+            .map(|obj| obj.version())
+            .unwrap_or_default();
+        self.try_get_past_object(
+            object_id,
+            version,
+            Some(SuiObjectDataOptions::bcs_lossless()),
+        )
+        .await
```

Comment:
If this is an api that is only intended for use with the replay tool, then lets adjust the return type to only be the BCS data of the object itself (base64 encoded because JSON though unfortunately).

We can also drop the extra call to "try_get_past_object" since the first call to `find_object_lt_or_eq_version` actually returns the raw object itself.
```suggestion
    ) -> RpcResult<Option<Object>> { // Return type probably needs to be massaged a bit for JSONRPC
        self
            .state
            .find_object_lt_or_eq_version(&object_id, &version)
            .await
            .map_err(Error::from)
```

Reply:
yeah, it was intentional to keep the same format and not to perform any jsonrpc conversions(slight overhead should be ok in this case). But ok, changed to `RpcResult<Option<SuiRawData>>`